### DISCUSSION TITLE
How do I setup an SPI interface, and communicate with my device?

### DISCUSSION THREAD

**Discussion by CoolSlimbo** [Rahix/avr-hal#441] [Q&A] [answered]
As the title says.

I've found it rather difficult to figure out how to set up a SPI device and communicate with it successfully.

From what I've been able to find, I've been unable to find anything. As in, nothing that directly helps someone find what they're looking for in regards to making an SPI interface and such.

With regard to this, how does one set up an SPI device and communicate with it successfully, using these libraries? (In my specific case, `arduino-hal`)

**Comment by Rahix** [comment, marked as answer]
(Converted this issue to a discussion).

Not sure why you closed the issue, maybe you already figured it out.

To document this for everyone else: There is an example for setting up SPI as part of the examples directory: https://github.com/Rahix/avr-hal/blob/main/examples/arduino-uno/src/bin/uno-spi-feedback.rs

**Reply by CoolSlimbo** [reply]
Hey. Just followed the example you linked!

Didn’t realise it existed until later, and realised it wasn’t needed (the issue).

Had to implement a different system anyways due to the one byte limit.
